README.md

```diff
@@ -1 +1,216 @@
+
+`<field>` elements have 1 attribute:
+1. fid: the OSSI field id
+
+
+Examples:
+
+    curl -d 'request=<command pbxName="uw01" cmdType="ossi" cmd="display time"/>' \
+    http://localhost:8000/v2/uw01
+
+
+    curl -d 'request=<command pbxName="uw01" cmdType="ossi" cmd="display station 12345">
+       <field fid="8005ff00"/>
+       <field fid="8003ff00">12345 Test</field>
+    </command>' \
+    http://localhost:8000/v2/uw01
+
+    curl -d 'request=<command pbxName="uw01" cmdType="ossi" cmd="list extension count 10">
+       <field fid="0001ff00"/>
+       <field fid="0002ff00"/>
+    </command>' \
+    http://localhost:8000/v2/uw01
+
+The v2 API will return a XML document with the `<command>` element and one or more of these child elements:
+- `<ossi_object>`
+- `<screen>`
+- `<error>`
+
+
+## Using pbx.definity in your own python code
+
+You can import pbxd.pbx.definity in your own python code if you want to have more
+control or don't need the pbxd web service.
+
+    python3 -m venv venv && . venv/bin/activate
+    pip install -r requirements.txt
+    pip install -e .
+
+Example script:
+
+    import pbxd.pbx.definity as definity
+    import json
+    import logging
+    import sys
+
+    logging.basicConfig(stream=sys.stdout, level=logging.DEBUG,
+        format='%(asctime)s %(levelname)-8s [%(filename)s:%(lineno)d] - %(message)s'
+    )
+
+    with open('pbxd_conf/pbxd_uw01_conf.json') as json_file:
+        config = json.load(json_file)
+
+    pbx = definity.Terminal(
+        config['connection_command'],
+        config['pbx_username'],
+        config['pbx_password']
+    )
+    pbx.connect()
+
+    fields = {'0003ff00': ''}
+    result = pbx.ossi_command('display time', fields)  # returns {"ossi_objects": [{"0003ff00": "2020"}]}
+    ossi = result['ossi_objects'][0]
+    print('The PBX says the year is {}'.format(ossi.get('0003ff00')))
+
+    vt220 = pbx.vt220_command('display time')
+    print('\n'.join(vt220['screens']))
+
+    pbx.disconnect()
+
+## Background
+
+The development of pbxd originated with a Perl module developed to support
+scripts for automating management of the Lucent G3r PBX in the earl 2000's.
+As more scripts were written and run there was duplicate code and multiple
+scripts accessing the PBX simultaneously. There were also more PBX systems at
+remote sites. Pbxd was created to share connections to a PBX, and enable
+multiple clients like web browsers to get bits of information from the PBX.
+
+Written in Perl as a CGI script it worked well enough for many years but given
+time everything changes. It is still a useful tool and the Avaya Communication
+Manager still supports the OSSI protocol so pbxd has been rewritting in Python
+with a JSON API.
+
+You can see a dscription of the original version here:
+http://tools.cac.washington.edu/2010/04/avaya-pbx-admin-web-service.html
```

Comment:
This is markdown, might as well use a [real link](http://tools.cac.washington.edu/2010/04/avaya-pbx-admin-web-service.html)